rhino/src/main/java/org/mozilla/javascript/regexp/UnicodeProperties.java

```diff
@@ -0,0 +1,437 @@
+    public static final byte SURROGATE = PRIVATE_USE + 1;
+    public static final byte LETTER = SURROGATE + 1;
+    public static final byte LOWERCASE_LETTER = LETTER + 1;
+    public static final byte MODIFIER_LETTER = LOWERCASE_LETTER + 1;
+    public static final byte OTHER_LETTER = MODIFIER_LETTER + 1;
+    public static final byte TITLECASE_LETTER = OTHER_LETTER + 1;
+    public static final byte UPPERCASE_LETTER = TITLECASE_LETTER + 1;
+    public static final byte MARK = UPPERCASE_LETTER + 1;
+    public static final byte SPACING_MARK = MARK + 1;
+    public static final byte ENCLOSING_MARK = SPACING_MARK + 1;
+    public static final byte NONSPACING_MARK = ENCLOSING_MARK + 1;
+    public static final byte NUMBER = NONSPACING_MARK + 1;
+    public static final byte DECIMAL_NUMBER = NUMBER + 1;
+    public static final byte LETTER_NUMBER = DECIMAL_NUMBER + 1;
+    public static final byte OTHER_NUMBER = LETTER_NUMBER + 1;
+    public static final byte PUNCTUATION = OTHER_NUMBER + 1;
+    public static final byte CONNECTOR_PUNCTUATION = PUNCTUATION + 1;
+    public static final byte DASH_PUNCTUATION = CONNECTOR_PUNCTUATION + 1;
+    public static final byte CLOSE_PUNCTUATION = DASH_PUNCTUATION + 1;
+    public static final byte FINAL_PUNCTUATION = CLOSE_PUNCTUATION + 1;
+    public static final byte INITIAL_PUNCTUATION = FINAL_PUNCTUATION + 1;
+    public static final byte OTHER_PUNCTUATION = INITIAL_PUNCTUATION + 1;
+    public static final byte OPEN_PUNCTUATION = OTHER_PUNCTUATION + 1;
+    public static final byte SYMBOL = OPEN_PUNCTUATION + 1;
+    public static final byte CURRENCY_SYMBOL = SYMBOL + 1;
+    public static final byte MODIFIER_SYMBOL = CURRENCY_SYMBOL + 1;
+    public static final byte MATH_SYMBOL = MODIFIER_SYMBOL + 1;
+    public static final byte OTHER_SYMBOL = MATH_SYMBOL + 1;
+    public static final byte SEPARATOR = OTHER_SYMBOL + 1;
+    public static final byte LINE_SEPARATOR = SEPARATOR + 1;
+    public static final byte PARAGRAPH_SEPARATOR = LINE_SEPARATOR + 1;
+    public static final byte SPACE_SEPARATOR = PARAGRAPH_SEPARATOR + 1;
+
+    // Binary property values
+    public static final byte TRUE = SPACE_SEPARATOR + 1;
+    public static final byte FALSE = TRUE + 1;
+
+    // Property Name Map (canonical names and aliases)
+    public static final Map<String, Byte> PROPERTY_NAMES =
+            Map.ofEntries(
+                    Map.entry("Alphabetic", ALPHABETIC),
+                    Map.entry("Alpha", ALPHABETIC),
+                    Map.entry("ASCII", ASCII),
+                    Map.entry("Case_Ignorable", CASE_IGNORABLE),
+                    Map.entry("CI", CASE_IGNORABLE),
+                    Map.entry("General_Category", GENERAL_CATEGORY),
+                    Map.entry("gc", GENERAL_CATEGORY),
+                    Map.entry("Script", SCRIPT),
+                    Map.entry("sc", SCRIPT),
+                    Map.entry("Hex_Digit", HEX_DIGIT),
+                    Map.entry("Hex", HEX_DIGIT),
```

Comment:
The way it's implemented corresponds to "AHex", Ascii hex digit.
`\p{Hex}` should also match e.g. https://www.compart.com/en/unicode/U+FF11